ang/civicase-base/providers/activity-forms.provider.js

```diff
@@ -0,0 +1,74 @@
+(function (_, angular) {
+  var module = angular.module('civicase-base');
+
+  module.provider('ActivityForms', ActivityFormsProvider);
+
+  /**
+   * Activity Forms Provider.
+   */
+  function ActivityFormsProvider () {
+    var activityForms = [];
+
+    this.$get = $get;
+    this.addActivityForms = addActivityForms;
+
+    /**
+     * Returns a `getActivityFormService` function that can be used to find
+     * right right form service for the given activity.
+     *
+     * @param {object} $injector the angular injector service.
+     * @returns {object} The Activity Forms service.
+     */
+    function $get ($injector) {
+      var activityFormServices = _.chain(activityForms)
+        .sortBy('weight')
+        .map('name')
+        .map(injectActivityFormService)
+        .value();
+
+      return {
+        getActivityFormService
+      };
+
+      /**
+       * @param {object} activity an activity object.
```

Comment:
Do we not get errors if description of the function is missing.

Reply:
It does not. I believe it only complains about the missing description when there are no params and returns statements. In this case, I think the `@returns` explains pretty well what the function does, otherwise there would be repetition between the description and the return.

What do you think?

Reply:
Its fine